docs/src/main/asciidoc/kafka-streams.adoc

```diff
@@ -722,3 +722,226 @@ For more details about the health information, see the
+		return input ->
+		        input
+                    .flatMapValues(value -> Arrays.asList(value.toLowerCase().split("\\W+")))
+                    .groupBy((key, value) -> value)
+                    .windowedBy(TimeWindows.of(5000))
+                    .count(Materialized.as("WordCounts-multi"))
+                    .toStream()
+                    .map((key, value) -> new KeyValue<>(null, new WordCount(key.key(), value, new Date(key.window().start()), new Date(key.window().end()))));
+    }
+
+	public static void main(String[] args) {
+		SpringApplication.run(WordCountProcessorApplication.class, args);
+	}
+----
+
+The input will be received from the input binding defined in the `KafkaStreamsProcessor` interface and the output will be sent to the output binding.
+
+If the processor does not send any data on the outbound, then this becomes a plain Consumer bean as below.
+
+[source]
+----
+
+	@Bean
+	public Consumer<KStream<?, String>> process() {
+		return input ->
+		        ....
+    }
+----
+
+
+Applications are free to define custom bindings and use that instead of the out of the box `KafkaStreamsProcessor` interface.
+
+==== Functions with multiple input bindings
+
+With `StreamListener`, we define multiple `Input` bindings and then later on use them as inputs in the method.
+With the functions approach, we still need to define those bindings in the binding interface. However, it cannot be used in the same way as in a `StreamListener` method.
+We use curried functions to represent multiple input destinations in the same processor.
+For instance, if a function has 2 inputs, the application define 2 partial functions in the function bean method. Lets see some examples.
+
+[source]
+----
+	@Bean
+    public Function<KStream<String, Long>,
+                    Function<KTable<String, String>, KStream<String, Long>>> process() {
+        return userClicksStream ->
+                (userRegionsTable ->
+                  (userClicksStream
+                        .leftJoin(userRegionsTable, (clicks, region) -> new RegionWithClicks(region == null ?
+                                        "UNKNOWN" : region, clicks),
+                                Joined.with(Serdes.String(), Serdes.Long(), null))
+                        .map((user, regionWithClicks) -> new KeyValue<>(regionWithClicks.getRegion(),
+                                regionWithClicks.getClicks()))
+                        .groupByKey(Serialized.with(Serdes.String(), Serdes.Long()))
+                        .reduce((firstClicks, secondClicks) -> firstClicks + secondClicks)
+                        .toStream()));
+    }
+----
+
+In the above function bean, there are 2 inputs and one input. The function that returns from the method takes a `KStream` as input, but if you look at the output that is another function
+which takes a `KTable` as its input. The output of this second function is a `KStream` which becomes the output of the procesor.
```

Comment:
It'd be useful to clarify how the input/output mapping is done:

Function 1: `Function<KStream<String, Long>` - KStream input; returns the output of "Function 2"
Function 2: `Function<KTable<String, String>, KStream<String, Long>>` - KTable input; returns `KStream<String, Long>`